Tchap/Modules/Settings/SettingsViewController.m

```diff
@@ -2661,4 +2708,33 @@ - (void)changePasswordAlertPresenterDidTapChangePasswordAction:(ChangePasswordAl
 
 #endif
 
+#pragma mark - MXKDocumentPickerPresenterDelegate
+
+- (void)documentPickerPresenterWasCancelled:(MXKDocumentPickerPresenter *)presenter
+{
+    self.documentPickerPresenter = nil;
+}
+
+- (void)documentPickerPresenter:(MXKDocumentPickerPresenter *)presenter didPickDocumentsAt:(NSURL *)url
+{
+    self.documentPickerPresenter = nil;
+    
+    if ([MXMegolmExportEncryption isMegolmKeyFile:url])
+    {
+        // Show the keys import dialog
+        self->importView = [[MXKEncryptionKeysImportView alloc] initWithMatrixSession:self.mainSession];
+        self->currentAlert = self->importView.alertController;
+        // WARNING: SettingsViewController should extends MXKViewController
+        // It's working because SettingsViewController has startActivityIndicator and
+        // stopActivityIndicator methods
```

Comment:
The right comment here would be:
"We have to change the signature of MXKEncryptionKeysImportView:showInViewController:toImportKeys:onComplete in the Kit,
- The first param must be `UIViewController <MXKViewControllerHandling>` instead of `MXKViewController`
- The `onComplete` block should return the error (nullable) to let the application handle it

Reply:
You may sum up this comment by using an issue number if you have created an issue in matrix-ios-kit

Reply:
Comment updated